dask_ml/model_selection/_incremental.py

```diff
@@ -347,3 +356,302 @@ def fit(
+# ----------------------------------------------------------------------------
+
+
+class BaseIncrementalSearchCV(DaskBaseSearchCV):
+    """Base class for estimators using the incremental `fit`.
+
+    Subclasses must implement the following abstract method
+
+    * _get_ids_and_args : Build the dict of
+       ``{model_id : (param_list, additional_calls)}``
+        that are eventually passed to `fit`
+    """
+
+    def __init__(
+        self,
+        estimator,
+        params,
+        test_size=0.15,
+        random_state=None,
+        scoring=None,
+        iid=True,
+        refit=True,
+        cv=None,
+        error_score="raise",
+        return_train_score=_RETURN_TRAIN_SCORE_DEFAULT,
+        scheduler=None,
+        n_jobs=-1,
+        cache_cv=True,
+    ):
+        # TODO: find the subset of sensible parameters.
+        self.params = params
+        self.test_size = test_size
+        self.random_state = random_state
+        super(BaseIncrementalSearchCV, self).__init__(
+            estimator,
+            scoring=scoring,
+            iid=iid,
+            refit=refit,
+            cv=cv,
+            error_score=error_score,
+            return_train_score=return_train_score,
+            scheduler=scheduler,
+            n_jobs=n_jobs,
+            cache_cv=cache_cv,
+        )
+
+    def _check_array(self, X, y, **kwargs):
+        """Validate the data arguments X and y.
+
+        By default, NumPy arrays are converted to 1-block dask arrays.
+
+        Parameters
+        ----------
+        X, y : array-like
+        """
+        if isinstance(X, np.ndarray):
+            X = da.from_array(X, X.shape)
+        if isinstance(y, np.ndarray):
+            y = da.from_array(y, y.shape)
+        X = check_array(X, **kwargs)
+        y = check_array(y, **kwargs)
+        return X, y
+
+    def _get_cv(self, X, y, **kwargs):
+        """CV-Split the arrays X and y
+
+        By default, :meth:`dask_ml.model_selection.train_test_split`
+        is used with ``self.test_size``. The test set is expected to
+        fit in memory on each worker machine.
+
+        Parameters
+        ----------
+        X, y : dask.array.Array
+        """
+        X_train, X_test, y_train, y_test = train_test_split(
+            X, y, test_size=self.test_size
+        )
+        return X_train, X_test, y_train, y_test
+
+    def _get_ids_and_args(self):
+        # type: () -> Dict[int, Tuple(List[Dict], Union[Callable, object])]
+        """Abstract method for generating the argumnets passed to `fit`.
+
+        Returns
+        -------
+        ids_and_args : Dict
+            Should have integers for keys and the following arguments
+
+            * param_list : List[Dict]
+            * additional_fit_calls : Union[Callable, object]
+                Objects with a ``fit`` method have the bound
+                ``additional_fit_calls.fit`` method passed through to `fit`.
+                (Other) callables are simply passed through.
```

Comment:
I would design things so that the `IncrementalRandomizedSearchCV` class has `remove_worst` as a method, rather than as a return value of `_get_ids_an_args`.  I think that the `remove_worst` method is likely to be the main point of variance between different classes, and so is something that people will probably expect to specify at the class level.

(also, just to be clear, we shouldn't actually publish a class with the `remove_worst` function as a genuine solution.  It's not a good solution)

I also don't understand why this returns a dictionary.